### ISSUE TITLE
이진 함수에서 Odds를 K개 Class로 일반화하는 과정이 이해가 가지 않습니다.

### ISSUE THREAD

**Issue by csmhub** [codingeverybody/codingyahac#1071]
Tensorflow 101의 16번째 강의를 듣는 과정에서 Softmax와 Sigmoid 함수의 차이점을 숙제로 받아 공부하고 있는 중 문제가 발생했습니다.

# 해결하고자 하는 문제
Softmax 함수를 유도하는 과정에서, 이진 함수의 Odds를 K개의 클래스가 있는 함수로 변환하는 부분이 이해가 가지 않습니다.
<img width="699" alt="스크린샷 2021-01-07 오후 4 17 37" src="https://user-images.githubusercontent.com/31226762/103863286-eabdef80-5103-11eb-9ef6-3063985210a4.png">

# 시도해본 방법
저는 Odds가 다른 확률비이므로, K개의 클래스가 있을 때 Odds는 아래처럼 나와야 한다고 생각했습니다. 
![image](https://user-images.githubusercontent.com/31226762/103863482-46887880-5104-11eb-8394-95af3b53a05a.png)
예를 들어 5개의 클래스가 있다고 치면, 첫 번째 클래스의 Odds는 "(첫 번째 클래스)/(나머지 클래스 확률 합)" 형태가 나와야 한다고 생각했습니다.

그런데 위 식은 K번째 클래스의 확률만 고려해서 Odds를 구하고 있는 것 처럼 보입니다. '그러면 그냥 K번째 클래스를 기준으로 나머지 확률의 비를 표현하게 되는 건가요?' '그렇다면 굳이 K번째일 이유도 있지 않나요?' 'K클래스의 확률이 심각하게 작거나 클 때 다른 확률을 표현하는 데에 문제가 생기지 않나요?' 등의 생각이 꼬리에 꼬리를 물고 올라왔습니다.

간단히 말해서, 첫 번째 사진에서 설명하는 원리를 잘 이해하지 못하겠습니다.

<a href="https://insomnia.tistory.com/12">여기</a>는 제가 공부하면서 필기해 놓은 내용입니다.

**Comment by jeuxdeau** [comment]
의문이 드신 점을 제가 제대로 이해한 것이 맞다면 y/(1-y)처럼 이것도 P(Cy)/(1-P(Cy)) 이런 비슷한 모양이 되어야 하지 않을까에서 출발하신 것 같아요

그래서 식 전개 과정 설명이 잘 되어있는 동영상을 찾아 봤는데 https://youtu.be/deNBpWGLYY4?t=440 (7분 20초쯤부터)

동영상 기준으로 (1) 구하고 싶은 것은 P(Cm)/(1-P(Cm))이고 (뒷부분에 나옵니다) (2) P(Ci)/P(Ck)는 (1)을 구하기 위해 만들어본 식 정도로 이해하시면 될 것 같습니다

=====

> '그러면 그냥 K번째 클래스를 기준으로 나머지 확률의 비를 표현하게 되는 건가요?' '그렇다면 굳이 K번째일 이유도 있지 않나요?' 'K클래스의 확률이 심각하게 작거나 클 때 다른 확률을 표현하는 데에 문제가 생기지 않나요?'

맨 마지막에 나온 식에서
![image](https://user-images.githubusercontent.com/22112344/103896130-d7774800-5134-11eb-8fa9-7e2089114151.png)

exp(t) 자리에 처음 시작했던 이걸 넣어서 전개해 보시면 됩니다
![image](https://user-images.githubusercontent.com/22112344/103896201-f1b12600-5134-11eb-9185-26512fe7eb32.png)

**Comment by csmhub** [comment]
제가 확실히 잘못 이해하고 있는 부분을 짚어주셨습니다... 감사합니다! 😀
